app/models/transformation_mapping.rb

```diff
@@ -0,0 +1,21 @@
+class TransformationMapping < ApplicationRecord
+  has_many :transformation_mapping_items, :dependent => :destroy
+
+  validates :name, :presence => true, :uniqueness => true
+
+  def self.select_mappings(vms)
+    resources = vms.each_with_object([]) do |vm, array|
+      array << vm.ems_cluster
```

Comment:
I'm not sure I know the schemea for vms, but can't we simply do something like
```
Cluster.where(vm_ids: vms)
```

Your code will go over every vm and fetch its cluster and strorage, this can be very slow on a very large number of vms